src/settings.rs

```diff
@@ -0,0 +1,20 @@
+use glib::variant::FromVariant;
+use glib::{IsA, ToVariant};
+use {Settings, SettingsExt};
+
+pub trait SettingsExtManual {
+    fn get<U: FromVariant>(&self, key: &str) -> Option<U>;
+
+    fn set<U: ToVariant>(&self, key: &str, value: &U) -> bool;
+}
+
+impl<O: IsA<Settings>> SettingsExtManual for O {
+    fn get<U: FromVariant>(&self, key: &str) -> Option<U> {
+        self.get_value(key)
+            .and_then(|val| FromVariant::from_variant(&val))
```

Comment:
Should probably return some kind of `Result` with a meaningful error type to distinguish "key not found" and "value of wrong type"

Reply:
Alright

Reply:
See the (new since yesterday) GValue API for some inspiration (or copy&paste)